web/app/scripts/filterbar/date-range-directive.js

```diff
@@ -2,54 +2,115 @@
     'use strict';
 
     /* ngInject */
-    function DateRangeField() {
+    function DateRangeField(DateLocalization) {
         var module = {
             restrict: 'A',
             require: ['^driver-filterbar', 'date-range-field'],
             templateUrl: 'scripts/filterbar/date-range.html',
             controller: 'dateRangeController',
             link: function(scope, elem, attrs, ctlArray) {
+                var calendar = null;
                 var filterLabel = '__dateRange';
                 var filterBarCtl = ctlArray[0];
                 var dateRangeCtl = ctlArray[1];
-                var dtRange = {};
-                scope.calendarOptions = {'format': filterBarCtl.dateTimeFormat};
+                var dtRange = {};  // internal min/max date strings, used for API
+                // scope.min and scope.max are localized strings for display
+
 
                 scope.$on('driver.filterbar:reset', function() {
                     init();
                 });
 
-                // restore previously set filter selection on page reload
+                /**
+                 * Restore a user's saved dates; this comes in from the localstorage as an
+                 * ISO-8601 string, so we need to convert that to a localized CDate so that
+                 * we can display it to the user.
+                 */
                 scope.$on('driver.filterbar:restored', function(event, filter) {
                     if (filter.label === filterLabel) {
-                        scope.dtMin = filter.value.min;
-                        scope.dtMax = filter.value.max;
+                        // The restored date will be an ISO-8601 string, so we need to convert
+                        // that to a Javascript Date, and then convert that to a localized CDate,
+                        // and then store the formatted string.
+                        dtRange.min = filter.value.min;
+                        dtRange.max = filter.value.max;
+                        if (dtRange.min) {
+                            var jsMin = moment(dtRange.min, moment.ISO_8601).toDate();
```

Comment:
Yeah, me too. It also didn't complain about that trailing comma, but it complained about some other stuff so it may not be configured quite correctly.